### PR TITLE
MD-06/filters into develop

### PR DESCRIPTION
**Pull request by DaanEKR**
# MD-06/filters into develop
Added filters on homepage

- Created search filter
- Created type filter
- Created gen filter
- Created caught filter

## Does this close any currently open issues?
#7 

## How to check:

- Search for a pokemon
- Use filters
- Use multiple filters

## Where has this been tested?
**Browsers:**
- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

## Needs 2 thumbs up

### CODE REVIEW

File: front-dex/src/app/components/card/card.component.ts [hunk 1]
```diff
@@ -8,12 +8,18 @@ import Pokemon from 'src/app/models/interfaces/Pokemon';
 })
 export class CardComponent implements OnInit {
   @Input() pokemon: Pokemon;
+  pokemonStorage = 'pokemon'
```
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.ts]: type?

File: front-dex/src/app/components/card/card.component.ts [hunk 2]
```diff
@@ -8,12 +8,18 @@ import Pokemon from 'src/app/models/interfaces/Pokemon';
...
 
   constructor() { }
 
   ngOnInit() {
   }
 
+  caughtCheck(pokemon) {
```
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.ts]: Type?

File: front-dex/src/app/components/card/card.component.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
     <section class="pokecard__header">
         <div class="pokecard__options">
             <app-favourite></app-favourite>
-            <app-caught class="pokecard__option-spacing"></app-caught>
+            <app-caught class="pokecard__option-spacing" (click)="caughtCheck(pokemon)"></app-caught>
```
Comment (koenkort) [inline front-dex/src/app/components/card/card.component.html:5]: Modifier should be '--spacing'

File: front-dex/src/app/components/card/card.component.ts (hunk 2 above)
Comment (kmensink95) [reply front-dex/src/app/components/card/card.component.ts]: And void or :: behind the function plz

File: front-dex/src/app/pages/home/home.component.scss [hunk 4]
```diff
@@ -44,3 +54,34 @@
     flex-direction: row;
     flex-wrap: wrap;
 }
+
+.check {
+    margin: 5px 0 0 20px;
+
+    &__input {
+        cursor: pointer;
+        position: absolute;
```
Comment (koenkort) [inline front-dex/src/app/pages/home/home.component.scss]: Why absolute?

**Pull request merged**